custom_components/elasticsearch/es_index_manager.py

```
@@ -55,8 +57,7 @@ def __init__(self, hass, config, gateway):
             self.datastream_name_prefix = config.get(CONF_DATASTREAM_NAME_PREFIX)
             self.datastream_namespace = config.get(CONF_DATASTREAM_NAMESPACE)
         else:
-            return
-
+            raise ElasticException("Unexpected index_mode: %s", self.index_mode)
```

This shouldn't ever happen, so I want to fail fast with an exception